### PR TITLE
Fix scroll issues

### PR DESCRIPTION
**Pull request by alejandraarri**
The footer has a position: fixed and therefore, the container is behind the footer. To respect that space and avoid overlapping I have placed a margin-bottom to the container with the same footer height.

Besides, I avoid the main scroll to appear when opening a modal.

Also, I fixed pagination that wasn't appearing when it should.

### Acceptance
Having more than 20 datasets, click “New map” button and check:
- That main scroll doesn’t appear.
- That the footer doesn’t overlap the dataset container.
- The pagination appears.

### Related Issue
https://github.com/CartoDB/cartodb/issues/14704, https://github.com/CartoDB/cartodb/issues/14703

### CODE REVIEW

File: lib/assets/javascripts/new-dashboard/components/Backbone/Dialog.vue [hunk 1]
```diff
@@ -17,6 +17,12 @@ export default {
     close () {
       this.$emit('close');
     }
+  },
+  beforeMount() {
+    document.body.classList.add('u-overflow-hidden');
+  },
+  beforeDestroy() {
```
[the remark this answers is not in the document]
Comment (alejandraarri) [reply lib/assets/javascripts/new-dashboard/components/Backbone/Dialog.vue]: Done!

File: lib/assets/javascripts/new-dashboard/components/Backbone/Dialog.vue [hunk 2]
```diff
@@ -17,6 +17,12 @@ export default {
     close () {
       this.$emit('close');
     }
+  },
+  beforeMount() {
```
[the remark this answers is not in the document]
Comment (alejandraarri) [reply lib/assets/javascripts/new-dashboard/components/Backbone/Dialog.vue]: Done!

File: assets/stylesheets/common/create/create_listing.scss [hunk 3]
```diff
@@ -15,13 +15,14 @@
...
+    padding-top: 0;
+    padding-bottom: 0;
+  }
 }
 
 .DatasetsPaginator .Pagination {
   justify-content: flex-end;
 }
 
-//Fix Pagination for add layer modal
```
Comment (jesusbotella) [inline assets/stylesheets/common/create/create_listing.scss]: What was this all about and why have we removed it? We might need that for Builder's modal 🤔

Comment (alejandraarri) [reply assets/stylesheets/common/create/create_listing.scss]: This was made to create the extra space we are now creating under the Dialog-body, so the footer didn't overlap the pagination. We should check in the staging instance that Builder is working ok.

Comment (jesusbotella) [reply assets/stylesheets/common/create/create_listing.scss]: Ok! 👍

**Comment by jesusbotella** [comment]
Everything looks good in staging! Better than it was before!

**Pull request merged**